lecture_5/homework/spec/helpers/reports/siege_report_spec.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Reports::SiegeReport, type: :helper do
+  subject(:clan) { create(:clan) }
+  subject(:building) { create(:building) }
+  subject(:building2) { create(:building) }
+  context(' valid data ') do
+    it 'set building siege_value to be 250' do
+      w1 = create(:warrior, :samurai, clan: clan, building: building)
+      w2 = create(:warrior, :samurai, clan: clan, building: building)
+      # Reports::SiegeReport.call(building: building)
+      expect(building.siege_ability).to eq(250)
+    end
+
+    it 'set building siege_value to be 500 after one warrior was destroyed' do
+      w1 = create(:warrior, :samurai, clan: clan, building: building)
+      w2 = create(:warrior, :samurai, clan: clan, building: building)
+      # Reports::SiegeReport.call(building: building)
+      w2.destroy!
+      expect(building.siege_ability).to eq(500)
+    end
+
+    it 'set building siege_value to be 0 after both warriors was destroyed' do
+      w1 = create(:warrior, :samurai, clan: clan, building: building)
+      w2 = create(:warrior, :samurai, clan: clan, building: building)
+      # Reports::SiegeReport.call(building: building)
+      w1.destroy
+      w2.destroy
+      expect(building.siege_ability).to eq(0)
+    end
+
+    # fail should 500 instead of 250
+    it 'set building siege_value to be 250 after building change' do
+      w1 = create(:warrior, :samurai, clan: clan, building: building)
+      w2 = create(:warrior, :samurai, clan: clan, building: building2)
+      Reports::SiegeReport.call(building: building)
+      w2.update(building: building)
+      expect(building.siege_ability).to eq(500)
+    end
+
+    # Fail got 250 instead of 500
+    it 'set building siege_value to be 500 after building change' do
+      w1 = create(:warrior, :samurai, clan: clan, building: building)
+      w2 = create(:warrior, :samurai, clan: clan, building: building)
+      Reports::SiegeReport.call(building: building)
+      w2.update!(building: building2)
+
+      expect(building.siege_ability).to eq(250)
+    end
+    # Fail got 500 instead of 0
```

Comment:
To może wynikać z tego, że dla building trzeba przeładować atrybuty i asocjacje z bazy za pomocą metody '.reload'